map/autoware_lanelet2_map_validator/docs/traffic_light/traffic_light_facing.md

```diff
@@ -0,0 +1,50 @@
+# traffic_light_facing
+
+## Validator name
+
+mapping.traffic_light.correct_facing
+
+## Recommended prerequisite validators
+
+- mapping.traffic_light.missing_referrers
+- mapping.traffic_light.regulatory_element_details
+
+## Feature
+
+This validator checks whether each `traffic_light` linestring is drawn with the correct direction, because the linestring direction describes the facing of the traffic_light.
+If the traffic light is facing to the viewer, the traffic light linestring must be drawn from the left point to the right point seen from the viewer.
+Note that this validator only check traffic lights whose subtype are `red_yellow_green`.
+This validator checks five types of issues.
+The former three issues are related to prerequisites to perform correct validation rather than direct validation results of the traffic light facing.
+The latter two issues mention to the traffic light facing.
+
+All output issues specify the traffic_light "linestring" or the traffic_light "regulatory_element" as the **primitive**, and the primitive ID will be specified as the **ID**.
+
+| Issue Code                     | Message                                                                       | Severity | Primitive  | Description                                                                                                                                                                                                                 | Approach                                                                                                                                                                                                                                      |
+| ------------------------------ | ----------------------------------------------------------------------------- | -------- | ---------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| TrafficLight.CorrectFacing-001 | "Lanelets referring this traffic_light have several divergent starting lines" | Info     | Linestring | A `traffic_light` subtype regulatory element may be referred by multiple lanelets. This warning appears when the starting line of those lanelets (which tends to be the same or similar) diverge too much.                  | This hardly happens, but maybe the referring lanelet is completely wrong or the traffic light cannot be seen from the starting edge of the referring lanelet.                                                                                 |
```

Comment:
Yes, like those kind of lanelets **but doesn't share the same starting edge**.
It is _very unnatural_ to happen but have to be promised not to happen because it is required in the calculation.